packages/neon_http_client/lib/src/client.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:cookie_store/cookie_store.dart';
+import 'package:http/http.dart' as http;
+import 'package:neon_http_client/src/interceptors/interceptors.dart';
+import 'package:universal_io/io.dart';
+
+/// An exception caused by an error in a [NeonClient].
+sealed class NeonClientException extends http.ClientException {
+  NeonClientException(super.message, [super.uri]);
+}
+
+/// An exception caused by a [HttpInterceptor].
+final class InterceptionFailure extends NeonClientException {
```

Comment:
```suggestion
final class InterceptionException extends NeonClientException {
```
Common naming scheme for exceptions

Reply:
I can see the point but later on we'll also need to wrap the `TimeoutException` and just calling it `TimeoutFailure` doesn't seem that bad.

Reply:
Hm we can do that for TimeoutException, but if it is not necessary here then it should follow the common naming scheme.